lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -774,9 +788,23 @@ protected static function compute_style_properties( $styles, $settings = array()
 			return $declarations;
 		}
 
+		$root_variable_duplicates = array();
+
 		foreach ( $properties as $css_property => $value_path ) {
 			$value = static::get_property_value( $styles, $value_path, $theme_json );
 
+			if ( strpos( $css_property, '--wp--style--root--' ) === 0 && static::ROOT_BLOCK_SELECTOR !== $selector ) {
+				continue;
+			}
+			// We don't support string values.
```

Comment:
Does this comment mean something these theme.json styles won't have any effect when I've set `"useRootVariables": true`?

```json
	"styles": {
		"spacing": {
			"padding": "25px"
		}
	},
```

Maybe a follow up might be to support it and generate the `top|right|bottom|left` CSS vars using the single string value?

Reply:
Yeah, I opted to leave this feature out for now because in the previous PR we got into a bit of discussion around the extent to which we should support stuff like `calc` in string values 😅 

I think it would be good to revisit in a follow-up! There's even an issue for it: #40132

Reply:
Great suggestion; done!